### PR TITLE
Update sample - query feature table

### PR DESCRIPTION
**Pull request by prupani-7**
## Description

PR to update the sample to use composable mapview from the toolkit

## Links and Data

Sample Epic: `runtime/kotlin/issues/3366`

## What To Review

Review updated files

## How to Test

Run the sample on the sample viewer or the repo.

### CODE REVIEW

File: query-feature-table/src/main/java/com/esri/arcgismaps/sample/queryfeaturetable/screens/MainScreen.kt
Comment (shubham7109) [inline query-feature-table/src/main/java/com/esri/arcgismaps/sample/queryfeaturetable/screens/MainScreen.kt:1]: Replace the `application` parameter from the MainScreen with the use of `LocalContext.current.applicationContext`

File: query-feature-table/src/main/java/com/esri/arcgismaps/sample/queryfeaturetable/components/MapViewModel.kt [hunk 1]
```diff
@@ -64,6 +61,17 @@ class MapViewModel(
...
+    // map used to display the feature layer
+    val map = ArcGISMap(BasemapStyle.ArcGISTopographic)
+
+    private var usaViewpoint = Viewpoint(
+        center = Point(-11e6, 5e6, SpatialReference.webMercator()),
+        scale = 1e8
+    )
+
+    // define a MapViewpointOperation to set the initial viewpoint and update it when the viewpoint changes
+    var mapViewpointOperation: MapViewpointOperation? by mutableStateOf(null)
```
Comment (shubham7109) [inline query-feature-table/src/main/java/com/esri/arcgismaps/sample/queryfeaturetable/components/MapViewModel.kt:73]: The `mapViewpointOperation` here does not need to be nullable, you could set it to the initial viewpoint. And remove it from the init block. 

```suggestion
    // define a mutable MapViewpointOperation and set the initial viewpoint
    var mapViewpointOperation: MapViewpointOperation by mutableStateOf(
        MapViewpointOperation.Set(usaViewpoint)
    )
```

File: query-feature-table/src/main/java/com/esri/arcgismaps/sample/queryfeaturetable/components/MapViewModel.kt [hunk 2]
```diff
@@ -113,28 +124,13 @@ class MapViewModel(
                 // get the extent of the first feature in the result to zoom to
                 val envelope = feature.geometry?.extent
                     ?: return@launch messageDialogVM.showMessageDialog("Error retrieving geometry extent")
-                // update the map view to set the viewpoint to the state geometry
-                mapViewState.stateGeometry = envelope
+                // update the MapViewpointOperation to set the geometry of the returned feature
```
Comment (shubham7109) [inline query-feature-table/src/main/java/com/esri/arcgismaps/sample/queryfeaturetable/components/MapViewModel.kt:127]: "set the geometry of the returned feature" doesn't sound right to me here. Maybe the comment could be: 

```suggestion
                // update the viewpoint to the bounding geometry of the returned feature
```

File: query-feature-table/src/main/java/com/esri/arcgismaps/sample/queryfeaturetable/components/MapViewModel.kt [hunk 3]
```diff
@@ -113,28 +124,13 @@ class MapViewModel(
...
         }
     }
 }
 
-/**
- * Class that represents the MapView state
- */
-class MapViewState {
-    // map used to display the feature layer
-    var arcGISMap: ArcGISMap by mutableStateOf(ArcGISMap(BasemapStyle.ArcGISTopographic))
```
Comment (shubham7109) [inline query-feature-table/src/main/java/com/esri/arcgismaps/sample/queryfeaturetable/components/MapViewModel.kt:136]: Extra line here

**Review by shubham7109 — CHANGES_REQUESTED** [review]
Sample works well! Just a few minor changes to make.

**Comment by prupani-7** [comment]
@shubham7109 Applied the suggestions, thanks! a re-review?

**Review by shubham7109 — APPROVED** [review]

**Review by changanxian — APPROVED** [review]
👍

**Pull request merged**